lectures/04_lecture/data_pipelines.ipynb

```diff
@@ -0,0 +1,1573 @@
+       "      <th>RandomForestClassifier(max_depth=80)</th>\n",
+       "      <td>0.901566</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=2)</th>\n",
+       "      <td>0.817866</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=6)</th>\n",
+       "      <td>0.865539</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=10)</th>\n",
+       "      <td>0.875183</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=20)</th>\n",
+       "      <td>0.869390</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>MLPClassifier(hidden_layer_sizes=40)</th>\n",
+       "      <td>0.877562</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                  score\n",
+       "LogisticRegression(multi_class='multinomial')  0.868228\n",
+       "RandomForestClassifier(max_depth=2)            0.816110\n",
+       "RandomForestClassifier(max_depth=5)            0.887573\n",
+       "RandomForestClassifier(max_depth=10)           0.899648\n",
+       "RandomForestClassifier(max_depth=20)           0.907510\n",
+       "RandomForestClassifier(max_depth=80)           0.901566\n",
+       "MLPClassifier(hidden_layer_sizes=2)            0.817866\n",
+       "MLPClassifier(hidden_layer_sizes=6)            0.865539\n",
+       "MLPClassifier(hidden_layer_sizes=10)           0.875183\n",
+       "MLPClassifier(hidden_layer_sizes=20)           0.869390\n",
+       "MLPClassifier(hidden_layer_sizes=40)           0.877562"
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sklearn.metrics import f1_score\n",
+    "\n",
+    "results = {}\n",
+    "for model in models_to_run:\n",
+    "\n",
+    "    # create instance of the class, feed the model to be tested\n",
+    "    clf = make_pipeline(categorical_cols, numerical_cols, model)\n",
+    "    clf.fit(X_train_wine, y_train_wine)\n",
+    "\n",
+    "    # store results in a dictionary \n",
+    "    results[str(model)] = f1_score(y_test_wine, clf.predict(X_test_wine), average='weighted')\n",
+    "pd.DataFrame.from_dict(results, orient='index', columns=['score'])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "An interesting observation is that the Neural Network does not seem to do much better than the Random Forest. Ofcourse, this might be different with proper hyperparameter tuning (see next week!), but something else must be mentioned. On tabular data (data that fit into an excel spreadsheet), Neural Networks might not be the best choice. A Random Forest, on the other hand, tends to have a good performance on tabular data."
```

Comment:
```suggestion
    "An interesting observation is that the Neural Network does not seem to do much better than the Random Forest. Of course, this might be different with proper hyperparameter tuning (see next week!), but something else must be mentioned. On tabular data (data that fit into an excel spreadsheet), Neural Networks might not be the best choice. A Random Forest, on the other hand, tends to have a good performance on tabular data."
```